### ISSUE TITLE
reproducing GloVe avg. emb results on STS

### ISSUE THREAD

**Issue by ozancaglayan** [UKPLab/sentence-transformers#1165]
Hello,

I was reading the recent simCSE paper which referred to your paper when reporting the Average GloVe embedding results for the STS benchmarks. I originally created the issue in their [repository](https://github.com/princeton-nlp/SimCSE/issues/87)  to ask but this may be a better place: How can I reproduce those numbers because by using the SentEval copy in simCSE's repository, I get very poor correlation numbers compared to your paper. 

Thanks

**Comment by nreimers** [comment]
The STS tasks (like STS12) are split in several sub-datasets. In the original STS12 workshop, the main evaluation method was to combine all datasets to a long list of pairs, compute the similarity score, and then compute the correlation to the gold scores.

SentEval deviates from this and computes the correlation on each sub-dataset individually, and then performs an average. This can have quite a big impact, either positive or negative, based on the model.

In the SBERT paper, I followed the recommendation from the original STS shared task, combine all datasets, and then compute one correlation score.

**Comment by ozancaglayan** [comment]
Thanks. This is what has been done in the simCSE paper referring to your ALL approach. They have their code implementing this approach but still I can't get your numbers. Do you have the code from your paper? If yes, could you open-source it?

**Comment by nreimers** [comment]
Have you tried the average glove embedding model provided in this repo? It removes stop-words, which improves the performance for avg. word embeddings quite a lot.

**Comment by ozancaglayan** [comment]
Sorry, checked every folder in the codebase but couldn't find that model here.

**Comment by nreimers** [comment]
https://www.sbert.net/docs/pretrained_models.html#average-word-embeddings-models

**Comment by 336655asd** [comment]
> https://www.sbert.net/docs/pretrained_models.html#average-word-embeddings-models

Yes, this model can reproduce the results close to your paper. Thanks a lot.

**Comment by batubb** [comment]
Yeah, I get 61.54. I wish there was a script for every result presented in the paper.
